File: elixir/bird-count/lib/bird_count.ex
defmodule BirdCount do
  def today([]), do: nil
  def today([today_count | _]), do: today_count

  def increment_day_count([]), do: [1]
  def increment_day_count([today_count | tail]), do: [today_count + 1 | tail]

  def has_day_without_birds?([]), do: false
  def has_day_without_birds?([count | _tail]) when count == 0, do: true
  def has_day_without_birds?([count | tail]), do: has_day_without_birds?(tail)

  def total([]), do: 0
  def total([count | tail]), do: count + total(tail)

  def busy_days([]), do: 0

  def busy_days([count | tail]) do
    cond do
      count >= 5 ->
        1 + busy_days(tail)

      true ->
        busy_days(tail)
    end
  end
end

File: ruby/bird-count/bird_count.rb
class BirdCount
  def self.last_week
    [0, 2, 5, 3, 7, 8, 4]
  end

  def initialize(birds_per_day)
    @bird_count = birds_per_day
  end

  def yesterday
    @bird_count[-2]
  end

  def total
    @bird_count.sum
  end

  def busy_days
    @bird_count.count {|num| num >= 5 }
  end

  def day_without_birds?
    @bird_count.any? {|num| num == 0}
  end
end
